### PR TITLE
Flannel integration enhancements and bug fixes

### PR DESCRIPTION
**Pull request by aanm**
Read per commit basis.

A follow up PR will be made scripts to cleanup BPF programs upon uninstall.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6726)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Review by vadorovsky — APPROVED** [review]

**Review by borkmann — APPROVED** [review]

File: daemon/daemon_main.go [hunk 1]
```diff
@@ -1102,7 +1102,7 @@ func runDaemon() {
 		if err != nil {
 			log.WithError(err).WithField("device", option.Config.FlannelMasterDevice).Fatal("Unable to set internal IPv4")
 		}
-		// d.attachExistingInfraContainers()
+		d.attachExistingInfraContainers()
```
Comment (ianvernon) [inline daemon/daemon_main.go]: What will happen when this code runs and the Cilium DaemonSet is not running with `hostPID: true`?

**Review by ianvernon — APPROVED** [review]

**Comment by aanm** [comment]
I know what the previous flake is, I'll make a PR afterwards

File: daemon/daemon_main.go (hunk 1 above)
Comment (aanm) [reply daemon/daemon_main.go]: Currently yes, I'll think on something to avoid this.

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

File: examples/kubernetes/1.10/cilium-crio-ds.yaml [hunk 2]
```diff
@@ -121,6 +121,12 @@ spec:
               key: ct-global-max-entries-other
               name: cilium-config
               optional: true
+        - name: CILIUM_PREALLOCATE_BPF_MAPS
```
Comment (ianvernon) [inline examples/kubernetes/1.10/cilium-crio-ds.yaml:124]: I'm surprised this was missing from the generated file before.

File: pkg/client/client.go [hunk 3]
```diff
@@ -86,14 +86,18 @@ func NewDefaultClientWithTimeout(timeout time.Duration) (*Client, error) {
...
 		case <-timeoutAfter:
-			return c, fmt.Errorf("Failed to create cilium agent client after %f seconds timeout: %s", timeout.Seconds(), err)
+			return nil, fmt.Errorf("Failed to create cilium agent client after %f seconds timeout: %s", timeout.Seconds(), err)
 		default:
 		}
 
 		c, err = NewDefaultClient()
 		if err == nil {
-			break
+			_, err := c.Daemon.GetConfig(nil)
```
Comment (ianvernon) [inline pkg/client/client.go:95]: Not-blocking: would be helpful to add a comment indicating that we make a request against the daemon here purely to see if it has connectivity (i.e., calling `GetConfig` vs. some other API call doesn't matter).

File: examples/kubernetes/addons/flannel/README.md [hunk 4]
```diff
@@ -4,8 +4,12 @@ Cilium integration with Flannel (EXPERIMENTAL)
 This directory contains the necessary scripts to run Cilium on top of your
 Flannel cluster.
 
-If you have your cluster already set up with Flannel will not need to install
-Flannel again.
+If you have your cluster already set up with Flannel you will not need to
+install Flannel again.
+
+This Cilium integration with Flannel was performed with Flannel 0.10.0. If you
+find any issues with previous Flannel versions please feel free to reach us out
```
Comment (ianvernon) [inline examples/kubernetes/addons/flannel/README.md:11]: Nit: reach out to us to help you

**Review by ianvernon — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Pull request merged**